e2e/tests/test_e2e.py

```diff
@@ -118,6 +118,61 @@ def test_enable_load_balancer(self):
         self.assertEqual(repl_svc_type, 'ClusterIP',
                          "Expected ClusterIP service type for replica, found {}".format(repl_svc_type))
 
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_lazy_spilo_update(self):
+        '''
+        Test lazy update for the Spilo image: operator changes a stateful set but lets pods run with the old image 
+        until they are recreated for reasons other than operator's activity. That works because the operator uses 
+        "onDelete" pod update policy for stateful sets.
+
+        The test covers:
+        1) enabling lazy upgrade in existing operator deployment
+        2) forcing the normal rolling upgrade by changing the operator configmap and restarting its pod
+        '''
+
+        k8s = self.k8s
+        pod0 = "acid-minimal-cluster-0"
+        pod1 = "acid-minimal-cluster-1"
+
+        # enable lazy update
+        patch_lazy_spilo_upgrade = {
+            "data": {
+                "enable_lazy_spilo_upgrade": "true",
+                "docker_image": "registry.opensource.zalan.do/acid/spilo-cdp-12:1.6-p16"
+            }
+        }
+        k8s.update_config(patch_lazy_spilo_upgrade)
+
+        # wait for sts update
+        time.sleep(60)
+
+        # restart the pod to get a container with the new image 
+        k8s.api.core_v1.delete_namespaced_pod(pod0, "default")
```

Comment:
can we make sure we delete the current replica pod and not simply the `-0` instance?

Reply:
it does not matter here, if we re-start master or replica. We simply need one newly launched pod to check that it indeed runs a new image as required by the semantics of lazy update